libtransmission/metainfo.c

```diff
@@ -590,6 +590,19 @@ static char const* tr_metainfoParseImpl(tr_session const* session, tr_info* inf,
 
     inf->isPrivate = i != 0;
 
+    /* source */
+    len = 0;
+    if (!tr_variantDictFindStr(infoDict, TR_KEY_source, &str, &len))
+    {
+        if (!tr_variantDictFindStr(meta, TR_KEY_source, &str, &len))
+        {
+            str = "";
+        }
+    }
+
+    tr_free(inf->source);
```

Comment:
We still need a call like this in `tr_metainfoFree()`, probably makes sense to put it right next to `comment` similar to the order of declaration in the struct

Reply:
Good catch. Thanks.